### PR TITLE
Configure Email provider warning when Novu email provider is active

### PR DESCRIPTION
**Pull request by djabarovgeorge**
### What change does this PR introduce?

The Novu Email integration is active by default for all clients (with a limit of 300 emails). We do show the Please configure Email provider to activate the channel warning banner under the Workflow Settings → Providers tab, even if the client didn't reach the emails sent count limit (300 emails sent).

Definition of done

Workflow Settings → Providers tab

* The Novu email provider instance is displayed in the providers list
* The warning banner Connect a provider for this channel should be only shown for the Email and SMS channels to the users that don't have any custom providers set up yet. See design below

I was not sure about the exact design therefore i added some question in Figma
https://www.figma.com/file/72BWZJMFf9aikgI7CPpohp/Workflow-create%2Fedit?type=design&node-id=505-313812&mode=design&t=kTqZeooKIZPyvmYb-0

image
image

### CODE REVIEW

File: apps/web/src/design-system/icons/arrows/CircleArrowRight.tsx
Comment (djabarovgeorge) [inline apps/web/src/design-system/icons/arrows/CircleArrowRight.tsx:1]: Removed the override, the same behavior remained.

image

File: apps/web/src/pages/integrations/components/NovuInAppForm.tsx
Comment (djabarovgeorge) [inline apps/web/src/pages/integrations/components/NovuInAppForm.tsx:1]: Related comment regarding this change.

https://github.com/novuhq/novu/pull/3908/files#r1282394657

File: apps/web/src/pages/templates/components/LackIntegrationAlert.tsx [hunk 1]
```diff
@@ -1,29 +1,30 @@
...
-const DoubleArrowRightStyled = styled(CircleArrowRight)`
-  cursor: pointer;
-`;
+type alertType = 'error' | 'warning';
 
-export function LackIntegrationError({
+export function LackIntegrationAlert({
   channelType,
   text,
+  type = 'error',
```
Comment (djabarovgeorge) [inline apps/web/src/pages/templates/components/LackIntegrationAlert.tsx:19]: Extended the component to switch between warning and error types, maybe in the future we will have success as well.

File: apps/web/src/pages/templates/components/LackIntegrationAlert.tsx [hunk 2]
```diff
@@ -76,6 +83,26 @@ const WarningMessage = styled.div`
...
+  background: ${({ backgroundColor }) => backgroundColor};
   border-radius: 7px;
 `;
+
+function alertTypeToDoubleArrowColor(type: alertType) {
+  switch (type) {
+    case 'warning':
+      return 'rgba(234, 169, 0, 1)';
+    default:
+      return 'undefined';
```
Comment (djabarovgeorge) [inline apps/web/src/pages/templates/components/LackIntegrationAlert.tsx:95]: at the moment the default is the error color so no need in mapping the default will handle it.

File: apps/web/src/pages/integrations/useProviders.ts [hunk 3]
```diff
@@ -2,20 +2,16 @@ import { useMemo } from 'react';
...
   providers,
-  ProvidersIdEnum,
   PushProviderIdEnum,
-  SmsProviderIdEnum,
 } from '@novu/shared';
 
 import { useIntegrations, useIsMultiProviderConfigurationEnabled } from '../../hooks';
 import type { IIntegratedProvider, IntegrationEntity } from './types';
 
-const NOVU_PROVIDERS: ProvidersIdEnum[] = [SmsProviderIdEnum.Novu, EmailProviderIdEnum.Novu];
```
Comment (djabarovgeorge) [inline apps/web/src/pages/integrations/useProviders.ts:17]: extracted to shared in order to reuse in ListProviders.tsx

**Review by scopsy — APPROVED** [review]
💪

**Pull request merged**